common/src/main/java/io/netty/util/DomainWildcardMappingBuilder.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+/**
+ * Builder that allows to build {@link Mapping}s that support
+ * <a href="https://tools.ietf.org/search/rfc6125#section-6.4">DNS wildcard</a> matching.
+ * @param <V> the type of the value that we map to.
+ */
+public class DomainWildcardMappingBuilder<V> {
+
+    private final V defaultValue;
+    private final Map<String, V> map;
+
+    /**
+     * Constructor with default initial capacity of the map holding the mappings
+     *
+     * @param defaultValue the default value for {@link Mapping#map(Object)} )} to return
+     *                     when nothing matches the input
+     */
+    public DomainWildcardMappingBuilder(V defaultValue) {
+        this(4, defaultValue);
```

Comment:
intentionally low default?

Reply:
yeah its what we use with our others as well...